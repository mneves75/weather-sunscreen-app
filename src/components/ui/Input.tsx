import React, { forwardRef } from 'react';
import { TextInput, TextInputProps } from 'react-native';

export type InputProps = TextInputProps & { invalid?: boolean };

const Input = forwardRef<TextInput, InputProps>(({ style, invalid, ...rest }, ref) => (
  <TextInput
    ref={ref}
    style={[
      {
        borderWidth: 1,
        borderRadius: 10,
        padding: 12,
        minHeight: 44,
        borderColor: invalid ? '#B00020' : '#CBD5E1',
        backgroundColor: '#fff',
      },
      style,
    ]}
    {...rest}
  />
));

Input.displayName = 'Input';

export default Input;
