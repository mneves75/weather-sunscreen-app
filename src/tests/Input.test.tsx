import React from 'react';
import { render, screen } from '@testing-library/react-native';
import Input from '../components/ui/Input';

describe('Input', () => {
  it('accepts text and marks invalid', () => {
    render(<Input accessibilityLabel="email" value="" onChangeText={() => {}} />);
    const el = screen.getByLabelText('email');
    expect(el).toBeTruthy();
  });

  it('applies invalid border when invalid=true', () => {
    const { toJSON } = render(
      <Input accessibilityLabel="field" invalid value="" onChangeText={() => {}} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
