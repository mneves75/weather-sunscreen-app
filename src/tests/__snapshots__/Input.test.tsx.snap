// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input applies invalid border when invalid=true 1`] = `
<TextInput
  accessibilityLabel="field"
  onChangeText={[Function]}
  style={
    [
      {
        "backgroundColor": "#fff",
        "borderColor": "#B00020",
        "borderRadius": 10,
        "borderWidth": 1,
        "minHeight": 44,
        "padding": 12,
      },
      undefined,
    ]
  }
  value=""
/>
`;
